src/OrbitGl/CaptureWindowTest.cpp

```diff
@@ -79,4 +81,193 @@ TEST_F(UnitTestCaptureWindow, SlidersRespondToMouseLeave) {
   EXPECT_FALSE(unit_test_vertical_slider->IsMouseOver());
 }
 
+class CaptureClientAppInterfaceFake : public orbit_capture_client::AppInterface {
+  [[nodiscard]] orbit_capture_client::CaptureClient::State GetCaptureState() const {
+    return orbit_capture_client::CaptureClient::State::kStopped;
+  }
+
+  [[nodiscard]] bool IsCapturing() const { return false; }
+
+  void StartCapture(){};
+  void StopCapture(){};
+  void AbortCapture(){};
+  void ToggleCapture(){};
+};
+
+class NavigationTestCaptureWindow : public CaptureWindow, public testing::Test {
+ public:
+  explicit NavigationTestCaptureWindow() : CaptureWindow(nullptr) {
+    capture_client_app_ = &capture_client_app_fake_;
+
+    Resize(600, 100);
+
+    capture_data_ = TrackTestData::GenerateTestCaptureData();
+    time_graph_ = std::make_unique<TimeGraph>(this, nullptr, &viewport_, capture_data_.get(),
+                                              &picking_manager_);
+
+    CreateSchedulerTimers();
+
+    // Prepare the TimeGraph to be fully zoomed out
+    UpdateChildrenPosAndSize();
```

Comment:
Could we just call `Draw` instead here? (I'm still trying to use public functions if possible. I'm sorry if that's what you already tried :see_no_evil: )

Reply:
Should we - yes. Could we - no, not yet, Orbit will crash and burn everything to the ground :-) There's still some places where we expect `app_ != nullptr`, and rendering functionality will fail without OpenGl initialization.

Reply:
I expected no less of an answer. :sweat_smile:

Reply:
As I wrote above, I've actually changed this to `PreRender()`, I think this is a bit clearer.